salt/modules/boto_kinesis.py

```diff
@@ -441,6 +441,26 @@ def reshard(stream_name, desired_size, force=False,
     return r
 
 
+def list_streams(region=None, key=None, keyid=None, profile=None):
+    '''
+    Return a list of all streams visible to the current account
+
+    CLI example::
+
+        salt myminion boto_kinesis.list_streams
+    '''
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    streams = []
+    ExclusiveStartStreamName = ''
+    while ExclusiveStartStreamName is not None:
+        args = {'ExclusiveStartStreamName': ExclusiveStartStreamName} if ExclusiveStartStreamName else {}
+        r = _execute_with_retries(conn, 'list_streams', **args)
+        r = r['result'] if r and r.get('result') else {}
+        streams += r.get('StreamNames', [])
+        ExclusiveStartStreamName = streams[-1] if r.get('HasMoreStreams', False) in (True, 'true') else None
+    return streams
```

Comment:
Error handling?

Reply:
Oop.  Sorry, I misread things and thought _execute_with_retries() was already handling that.  Just pushed an improved version :)